File: solutions/27.remove-element/src/two_pointers.rs
/*
 * @lc app=leetcode.cn id=27 lang=rust
 *
 * https://leetcode.cn/problems/remove-element/
 */

pub struct Solution;

// @lc code=start
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut left = 0;
        for right in 0..nums.len() {
            if nums[right] != val {
                nums[left] = nums[right];
                left += 1;
            }
        }
        left as i32
    }
}
// @lc code=end

File: solutions/167.two-sum-ii-input-array-is-sorted/src/two_pointers.ts
/*
 * @lc app=leetcode.cn id=167 lang=typescript
 *
 * https://leetcode.cn/problems/two-sum-ii-input-array-is-sorted/
 */

// @lc code=start
function twoSum(nums: number[], target: number): number[] {
    let left = 0;
    let right = nums.length - 1;
    while (left < right) {
        const sum = nums[left] + nums[right];
        if (sum < target) {
            left += 1;
        } else if (sum > target) {
            right -= 1;
        } else {
            return [left + 1, right + 1];
        }
    }
    return [-1, -1];
}
// @lc code=end

export { twoSum };
